near/kycdao-ntnft/src/lib.rs

```diff
@@ -1,21 +1,66 @@
 use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
-use near_sdk::collections::{LookupMap, LazyOption};
+use near_sdk::collections::{LookupMap, LazyOption, UnorderedMap};
 use near_sdk::*;
-use near_sdk::env::keccak256;
+use near_sdk::env::{keccak256, block_timestamp};
 use near_contract_standards::upgrade::Ownable;
 use near_contract_standards::ntnft::{Token, NTNFT, TokenId};
 use near_contract_standards::ntnft::metadata::*;
 
+use serde::{Serialize, Deserialize};
+
 type MintAuthorizationCode = u32;
 
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize)]
+pub struct Status {
+    /// shows if the token is revoked by the issuer
+    pub is_revoked: bool,
+    /// expiry timestamp
+    pub expiry: Option<u64>,
+}
+
+impl Status {
+    pub fn is_valid(&self) -> bool {
```

Comment:
Just noting - I guess because Rust standard is for snail case and Solidity is sort of camel case the interfaces won't be identical. Not a big deal.

Reply:
Yeah unfortunately that's how the languages are defined, we can't do anything about it, but hopefully it won't cause any confusion.